sympy/ntheory/residue_ntheory.py

```diff
@@ -742,6 +743,45 @@ def _nthroot_mod1(s, q, p, all_roots):
         return res
     return min(res)
 
+def _nthroot_mod_composite(a, n, m):
+    """
+    Find the solutions to ``x**n = a mod m`` when m is not prime.
+    """
+    from sympy.ntheory.modular import crt
+    f = factorint(m)
+    dd = {}
+    for p, e in f.items():
+        tot_roots = set()
+        if e == 1:
+            tot_roots.update(nthroot_mod(a, n, p, True) or [])
+        else:
+            for root in nthroot_mod(a, n, p, True) or []:
+                diff = (n * pow(root, n - 1)) % p
```

Comment:
```suggestion
                rootn = pow(root, n)
                diff = (rootn//root*n) % p
```

Reply:
root can be zero , so its better to use pow(root, n-1) and avoid division by root. I have made the changes.